### DISCUSSION TITLE
How to add scripts that have js codes to the HTML header?

### DISCUSSION THREAD

**Discussion by GoodluckH** [remix-run/remix#7356] [Q&A]
I'm trying to integrate a widget to my Remix site.

According to the widget provider: "It must be loaded only once your on pages. Place the next code before the closing HTML </head> tag."

Here's what the script looks like:

```html
<script>
  !function(){'use strict';var e=['debug','destroy','do','help','identify','is','off','on','ready','render','reset','safe','set'];if(window.noticeable)console.warn('Noticeable SDK code snippet loaded more than once');else{var n=function(e){return function(){var n=Array.prototype.slice.call(arguments);return n.unshift(e),t.push(n),t}},t=window.noticeable=window.noticeable||[];!function(){for(var o=0;o<e.length;o++){var r=e[o];t[r]=n(r)}}(),function(){var e=document.createElement('script');e.async=!0,e.src='https://sdk.noticeable.io/l.js';var n=document.head;n.insertBefore(e,n.firstChild)}()}}();
</script>
```

Initially, I was just trying to copy and paste the above in the `root.tsx` file. However, it seems like I cannot really put codes in it.

I was reading about External Scripts but still very unclear about how to properly load the script.

**Comment by anuragnandi** [comment]
Did you figure out the solution? I have the same problem when trying to load the google adsense scripts

**Reply by kiliman** [reply]
The same way would do it for any other React project.
```ts
<script
  dangerouslySetInnerHTML={{
    __html: `your script here`,
  }}
/>
```

**Reply by anuragnandi** [reply]
Thanks @kiliman , I have tried this It does not work for some reason, I need to link the adSense code.. it is something like this:

```
   <script
              async
              src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js?client=ca-pub-xxxxxxxxxx"
              crossOrigin="anonymous"
            />
            
```
            how do I add the above script, I see that the script is mounted but not executed.. even if it works, do not see the ads throughout the app, it is very instable
